test/replication-luatest/bootstrap_strategy_test.lua

```diff
@@ -455,3 +457,240 @@ g_config_fail.test_bad_uri_or_uuid = function()
         })
     end
 end
+
+local g_supervised = t.group('gh-8509-bootstrap-strategy-supervised')
+
+local server2_admin
+local SOCKET_TIMEOUT = 5
+
+local function make_bootstrap_leader_initial(sockname)
+    local sock, err = socket.tcp_connect('unix/', sockname)
+    t.assert_equals(err, nil, 'Connection successful')
+    local greeting = sock:read(128, SOCKET_TIMEOUT)
+    t.assert_str_contains(greeting, 'Tarantool', 'Connected to console')
+    t.assert_str_contains(greeting, 'Lua console', 'Connected to console')
+    sock:write('box.ctl.make_bootstrap_leader()\n', SOCKET_TIMEOUT)
+    local response = sock:read(8, SOCKET_TIMEOUT)
+    t.assert_equals(response, '---\n...\n', 'The call succeeded')
+end
```

Comment:
You should probably close the socket before returning from the function.

Reply:
Ok, done, although socket's gc metamethod closes it anyway.
```diff
diff --git a/test/replication-luatest/bootstrap_strategy_test.lua b/test/replication-luatest/bootstrap_strategy_test.lua
index b87614f36..d870fc4bf 100644
--- a/test/replication-luatest/bootstrap_strategy_test.lua
+++ b/test/replication-luatest/bootstrap_strategy_test.lua
@@ -463,6 +463,7 @@ local function make_bootstrap_leader_initial(sockname)
     sock:write('box.ctl.make_bootstrap_leader()\n', SOCKET_TIMEOUT)
     local response = sock:read(8, SOCKET_TIMEOUT)
     t.assert_equals(response, '---\n...\n', 'The call succeeded')
+    sock:close()
 end
 
 g_supervised.after_each(function(cg)
```